.github/workflows/unit_tests_with_nightly_puppet_gem.yaml

```diff
@@ -45,7 +45,7 @@ jobs:
 
       - name: Install the latest nightly build of puppet${{ matrix.puppet_version }} gem
         run: |
-          curl http://nightlies.puppet.com/downloads/gems/puppet${{ matrix.puppet_version }}-nightly/${{ matrix.gem_file }} --output puppet.gem
+          curl --location --retry 5 http://nightlies.puppet.com/downloads/gems/puppet${{ matrix.puppet_version }}-nightly/${{ matrix.gem_file }} --output puppet.gem
           gem install puppet.gem -N
```

Comment:
Maybe we could check if the provided file is a gem file (I think it's identified as tar archive by the file command)